tools/anthosbm-ansible-module/README.md

```diff
@@ -0,0 +1,841 @@
+    workstation:
+      hosts:
+        10.200.0.7:
+    cp_nodes:
+      hosts:
+        10.200.0.2:
+        10.200.0.3:
+        10.200.0.4:
+    worker_nodes:
+      hosts:
+        10.200.0.5:
+        10.200.0.6:
+```
+
+
+
+#### 5.6.2 Standalone Cluster Variables
+
+You need to set the _cluster\_type_ and _cluster\_name_ variables in the variable file.  Below is a sample variable file for the standalone cluster using Ubuntu nodes. Replace the values enclosed in square brackets ([]) with actual values relevant to your GCP project and the cluster.
+
+
+
+*vars/anthos_vars.yml*
+
+```
+login_user: anthos
+login_user_group: anthos
+login_user_home: /home/anthos
+os_type: "ubuntu"
+ws_docker: "yes"
+gcloud_sdk: "yes"
+kubectl_tool: "yes"
+bmctl_tool: "yes"
+service_accounts: "yes"
+bmctl_download_url: gs://anthos-baremetal-release/bmctl/1.7.0/linux-amd64/bmctl
+bmctl_workspace_dir: bmctl-workspace
+gcp_sa_key_dir: /home/anthos/gcp_keys
+local_gcr_sa_name: anthos-gcr-svc-account
+local_connect_agent_sa_name: connect-agent-svc-account
+local_connect_register_sa_name: connect-register-svc-account
+local_cloud_operations_sa_name: cloud-ops-svc-account
+ssh_private_key_path: /home/anthos/.ssh/id_rsa
+project_id: [PROJECT_ID]
+location: [REGION]
+cluster_name: [CLUSTER_NAME]
+cluster_type: standalone
+max_pod_per_node: 250
+container_runtime: docker
+app_logs: false
+pod_cidr: 192.168.0.0/16
+service_cidr: 10.96.0.0/12
+cp_vip: 10.200.0.47
+ingress_vip: 10.200.0.48
+lb_address_pool: 
+- 10.200.0.48/28
+admin_kubeconfig_path: 
+cgw_members: 
+- [email id of IAM user or service account]
+```
+
+The variables *admin_kubeconfig_path* is not used by the hybrid cluster.
+
+
+#### 5.6.3 Create Standalone Cluster
+
+Run below command from workstation node (ansible should be installed on the workstation node) to create the cluster.
+
+
+```
+cd <REPOSITORY_ROOT>/ansible
+ansible-playbook anthos.yml
+```
+
+
+
+## 6. Connect Gateway Configuration
+
+You can use Connect Gateway for connecting to the registered clusters and run commands to monitor the workload. You can read more about Connect Gateway [here](https://cloud.google.com/anthos/multicluster-management/gateway).
+
+
+### 6.1 Configure Connect Gateway
+
+You can configure the Connect Gateway with**connect-gateway** Ansible role. 
+
+
+*anthos.yml*
+
+```
+    - role: connect-gateway
+```
+
+
+This role requires the below variable that contains a list of users and/or service accounts that can connect to the cluster through Connect Gateway.
+
+
+*vars/anthos_vars.yml*
+
+```
+cgw_members: 
+- user:[USER_EMAIL_ID]
+- serviceAccount:[SERVICE_ACCOUNT_EMAIL_ID]
+```
+
+
+
+### 6.2 Validate Connect Gateway
+
+Open Cloud Shell from the GCP console after logging in using your Google account.
+
+
+```
+gcloud alpha container hub memberships get-credentials [CLUTER_NAME] --project [PROJECT_ID]
+```
+
+
+where `[CLUSTER_NAME]` is the name of the Anthos cluster and `[PROJECT_ID]` is the Google Cloud Project ID.
```

Comment:
Does the connect gateway need special whitelisting since it's in Alpha ?

Reply:
The Connect Gateway is in Pre-GA and does not require special whitelisting. The documentation (https://cloud.google.com/anthos/multicluster-management/gateway) is also available publicly now.